Firestore/Source/Remote/FSTDatastore.mm

```diff
@@ -305,20 +305,19 @@ - (void)invokeRPCWithFactory:(GRPCProtoCall * (^)(void))rpcFactory
   // TODO(mikelehen): We should force a refresh if the previous RPC failed due to an expired token,
   // but I'm not sure how to detect that right now. http://b/32762461
   _credentials->GetToken(
-      /*force_refresh=*/false,
-      [self, rpcFactory, errorHandler](Token result, const int64_t error_code,
-                                       const absl::string_view error_msg) {
-        NSError *error = util::WrapNSError(error_code, error_msg);
+      /*force_refresh=*/false, [self, rpcFactory, errorHandler](util::StatusOr<Token> result) {
+        NSError *error = util::WrapNSError(result.status());
         [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
           if (error) {
             errorHandler(error);
           } else {
             GRPCProtoCall *rpc = rpcFactory();
+            const Token &token = result.ValueOrDie();
```

Comment:
Just my 1 cent, check `result` rather than `error` in the condition is safer to call `result.ValueOrDie()`. Then we only assume `statusor`'s logic rather than also depending on `MakeNSError`'s.

Reply:
(See Gil's comment / done.)